index.bs

```diff
@@ -587,6 +587,44 @@ See also:
 * [[#secure-context]]
 * [[#consent]]
 
+<h3 id=text-formats>Make textual formats for people</h3>
+
+Design textual formats to be both produced and consumed by people.
+
+People who are presented with a textual format will
+expect to be able to use a text editor
+to produce or modify content.
+People who edit text will introduce a range of errors, but
+could struggle to identify and fix those errors.
+
+People expect some amount of flexibility in terms of how their edits are processed.
+Allowing flexibility in whitespace, quoting, delimiters, and other syntactic elements ensures
+that content is still comprehensible.
+
+<div class=example>
+The JSON format is an example of a format that appears to be flexible,
+but it lacks many of these basic usability amenities.
+In JSON there is:
+no commenting capability,
+objects and arrays cannot have trailing commas, and
+quotes are mandatory for object keys.
+This makes JSON prone to syntactic errors as a result of manual edits.
+</div>
+
+Containing the scope of a format that is affected by a syntax error
+could improve robustness and human usability.
+This requires that specifications fully define processing and
+error handling rules so that errors are handled in the same way.
```

Comment:
```suggestion
This requires that specifications define the data model first, and then fully define the processing and
error handling rules so that any arbitrary input byte stream will result in a consistent representation of the data.
```

Reply:
```suggestion
This requires that specifications fully define processing and
error handling so that all inputs result in consistent outcomes.
```

That's a lot of words.  Defining a data model is secondary so we can probably leave that out.  The key here is just that every implementation will produce the same output when given the same input.

It's not the case that arbitrary inputs always produce a good outcome, but that they are handled the same way by every implementation.